packages/client/hmi-client/src/components/drilldown/tera-ops-preview.vue

```diff
@@ -0,0 +1,113 @@
+<template>
+	<div class="preview">
+		<div class="content-container">
+			<header ref="header">
+				<h5>Preview</h5>
+				<Dropdown
+					v-if="options"
+					class="output-dropdown"
+					v-model="selectedValue"
+					:options="options"
+					@update:model-value="emit('change-output')"
+				></Dropdown>
+			</header>
+			<main :style="{ height: mainHeight }">
+				<slot />
+			</main>
+		</div>
+		<footer>
+			<Button v-if="canSaveAsset" outlined label="Save Model" @click="emit('save-asset')"></Button>
+			<div class="right-aligned-buttons">
+				<Button outlined label="Cancel" @click="emit('cancel')"></Button>
+				<Button label="Apply Changes and Close" @click="emit('apply-changes')"></Button>
+			</div>
+		</footer>
+	</div>
+</template>
+
+<script setup lang="ts">
+import { isEmpty } from 'lodash';
+import Button from 'primevue/button';
+import Dropdown from 'primevue/dropdown';
+import { nextTick, onMounted, onUpdated, ref, watch } from 'vue';
+
+const props = defineProps<{
+	options?: string[]; // subject to change based on how we want to pass in output data
+	// selectedOutput: string
+	value?: string;
+	canSaveAsset?: boolean;
+}>();
+
+const emit = defineEmits(['cancel', 'apply-changes', 'save-asset', 'change-output']);
+
+const selectedValue = ref<string>();
+
+const mainHeight = ref();
+const header = ref();
+
+onMounted(() => {
+	if (!isEmpty(props.options)) {
+		selectedValue.value = props.options?.[0];
+	}
+});
+
+onUpdated(async () => {
+	await nextTick();
+	const headerHeight = header.value?.offsetHeight;
+	mainHeight.value = `calc(100% - ${headerHeight}px)`;
+});
+
+watch(
+	() => props.value,
+	() => {
+		selectedValue.value = props.value;
+	}
+);
```

Comment:
Why not just use the props directly instead of creating this `const selectedValue = ref<string>();` ?

Reply:
I was following a bad way to use v-model on a custom component, will fix this up

Reply:
this is changed:
`tera-drilldown-preview`
```
<Dropdown
  v-if="options"
  class="output-dropdown"
  :model-value="output"
  :options="options"
  @update:model-value="emit('update:output', $event)"
 ></Dropdown>
```

When used
```
<tera-drilldown-preview :options="outputs" v-model:output="selectedOutput">
</tera-drilldown-preview>
```